components/blitz/src/omero/gateway/LoginCredentials.java

```diff
@@ -51,15 +57,74 @@
     /** ID of the group if not the default group of the user should be used */
     private long groupID = -1;
 
+    /** The connection argument.*/
+    private ImmutableList<String> args;
+
     /**
      * Creates a new instance
      */
     public LoginCredentials() {
         user = new UserCredentials();
         server = new ServerInformation();
+        args = null;
     }
 
     /**
+     * Creates a new instance.
+     * @param args The connection arguments.
+     */
+    public LoginCredentials(String[] args)
+    {
+        this();
+        if (args == null) {
+            throw new IllegalArgumentException("No connection arguments");
+        }
+        this.args = ImmutableList.copyOf(args);
+    }
+
+    /**
+     * Creates a new instance with the given credentials and default port
+     * 
+     * @param username
+     *            The username
+     * @param password
+     *            The password
+     * @param host
+     *            The server hostname
+     */
+    public LoginCredentials(String username, String password, String host) {
+        this(username, password, host, omero.constants.GLACIER2PORT.value);
```

Comment:
I'm not sure I follow, @mtbc. If you're saying is it necessary to know whether or not to pass 4064 or 4063 as the default, I'd say we always pass 4064. The handshake with the server will then tell if dropping down to 4063 is possible.